cpp/inline/mcts/ActionSelector.inl

```diff
@@ -38,6 +38,7 @@ inline ActionSelector<Game>::ActionSelector(const ManagerParams& params,
     edge_t* edge = node->get_edge(i);
     P(i) = edge->adjusted_policy_prior;
     E(i) = edge->E;
+    chance_dist[i] = edge->chance_prob;
```

Comment:
Think about this. Why, on chance-nodes, are you executing all this other crap in the `ActionSelector` constructor? It makes no sense. This is why the branch should happen before constructing the `ActionSelector`.